### ISSUE TITLE
Error: Download of zip failed - File size on server is too small...

### ISSUE THREAD

**Issue by oleteacher** [danielewood/sierra-wireless-modems#46]
Tried running script from terminal this evening and getting failed message when it gets to part where downloading firmware.

Ran "feeling brave" (wget https://raw.githubusercontent.com/danielewood/sierra-wireless-modems/master/autoflash-7455.sh && sudo bash autoflash-7455.sh) and ran local with same results. Here is portion of output:

```
OK
AT!IMAGE?
TYPE SLOT STATUS LRU FAILURES UNIQUE_ID   BUILD_ID
FW   1    EMPTY  0   0 0                  
FW   2    EMPTY  0   0 0                  
FW   3    EMPTY  0   0 0                  
FW   4    EMPTY  0   0 0                  
Max FW images: 4
Active FW image is at slot 255

TYPE SLOT STATUS LRU FAILURES UNIQUE_ID   BUILD_ID
Max PRI images: 50


OK
AT
OK
AT!IMAGE=0
OK
AT!IMAGE?
TYPE SLOT STATUS LRU FAILURES UNIQUE_ID   BUILD_ID
FW   1    EMPTY  0   0 0                  
FW   2    EMPTY  0   0 0                  
FW   3    EMPTY  0   0 0                  
FW   4    EMPTY  0   0 0                  
Max FW images: 4
Active FW image is at slot 255

TYPE SLOT STATUS LRU FAILURES UNIQUE_ID   BUILD_ID
Max PRI images: 50


OK
Waiting for modem to reboot...
---
Download of zip failed.
File size on server is too small, something is wrong, exiting...
Attempted download URL was: https://source.sierrawireless.com/~/media/support_downloads/airprime/74xx/fw/7455/zip
curl info:
HTTP/1.1 302 Found
Cache-Control: private
Content-Length: 148
Content-Type: text/html; charset=utf-8
Location: /sitecore/service/notfound.aspx
Server: Microsoft-IIS/10.0
Access-Control-Allow-Origin: *
Access-Control-Allow-Headers: Origin, X-Requested-With, Content-Type, Accept
Date: Fri, 26 Jun 2020 04:55:46 GMT
Set-Cookie: BIGipServer~SWI_CORP_WWW~SWI_CORP_CAKEL-HTTPS=928711690.47873.0000; path=/; Httponly; Secure
```
Tried on two different modems and using the same computer as always (never had issue). I am just not yet advanced enough to solve the issue, anyone that can help appreciated.

**Comment by oleteacher** [comment]
Thanks for confirming you get same error. I too am limited in knowledge, time to enroll in a course:)

Have been trying to find a zip file on sierra site, but all I come up with is:

https://source.sierrawireless.com/-/media/support_downloads/airprime/74xx/fw/7455/swi9x30c_02,-d-,33,-d-,03,-d-,00_generic_002,-d-,072_000.ashx

When you download above, it is the current firmware and in zip format.

Looking at line 304 in autoflash-7455.sh wondering if file size check is failing?

**Comment by theo400** [comment]
> Thanks @da6driver for confirming you get same error. I too am limited in knowledge, time to enroll in a course:)
> 
> Have been trying to find a zip file on sierra site, but all I come up with is:
> 
> https://source.sierrawireless.com/-/media/support_downloads/airprime/74xx/fw/7455/swi9x30c_02,-d-,33,-d-,03,-d-,00_generic_002,-d-,072_000.ashx
> 
> When you download above, it is the current firmware and in zip format.
> 
> Looking at line 304 in autoflash-7455.sh wondering if file size check is failing?

I got this to work with manually updating the file name and location.  It won't auto update to the latest, but does work for the latest version on the site right now.

**Comment by theo400** [comment]
> > Thanks @da6driver for confirming you get same error. I too am limited in knowledge, time to enroll in a course:)
> > Have been trying to find a zip file on sierra site, but all I come up with is:
> > https://source.sierrawireless.com/-/media/support_downloads/airprime/74xx/fw/7455/swi9x30c_02,-d-,33,-d-,03,-d-,00_generic_002,-d-,072_000.ashx
> > When you download above, it is the current firmware and in zip format.
> > Looking at line 304 in autoflash-7455.sh wondering if file size check is failing?
> 
> I got this to work with manually updating the file name and location. It won't auto update to the latest, but does work for the latest version on the site right now.

SWI9X30C_ZIP="${SWI9X30C_ZIP^^}"'zip

To

SWI9X30C_ZIP='SWI9X30C_02.33.03.00_Generic_002.072_000.zip'

**Comment by oleteacher** [comment]
Thank you @theo400 for sharing your solution.

**Comment by danielewood** [comment]
Resolved by #51

**Comment by oleteacher** [comment]
Your a champ Daniel, thank you!
